services/optimization.js

```diff
@@ -0,0 +1,162 @@
+'use strict';
+
+var v = require('./service-helpers/validator');
+var createServiceFactory = require('./service-helpers/create-service-factory');
+var objectClean = require('./service-helpers/object-clean');
+var stringifyBooleans = require('./service-helpers/stringify-booleans');
+
+/**
+ * Optimization API service.
+ *
+ * Learn more about this service and its responses in
+ * [the HTTP service documentation](https://www.mapbox.com/api-documentation/#optimization).
+ */
+var Optimization = {};
+
+/**
+ * Get a duration-optimized route.
+ *
+ * Please read [the full HTTP service documentation](https://www.mapbox.com/api-documentation/#optimization)
+ * to understand all of the available options.
+ *
+ * @param {Object} config
+ * @param {'driving'|'walking'|'cycling'} [config.profile="driving"]
+ * @param {Array<OptimizationWaypoint>} config.waypoints - An ordered array of [`OptimizationWaypoint`](#optimizationwaypoint) objects, between 2 and 12 (inclusive).
+ * @param {Array<'duration'|'distance'|'speed'>} [config.annotations] - Specify additional metadata that should be returned.
+ * @param {'any' | 'last'} [config.destination="any"] Returned route ends at `any` or `last` coordinate.
```

Comment:
Nitpicky formatting points — spaces around `|` and lack of `-` between name and description.